### PR TITLE
initial commit: test run on haxby dataset

### PR DESCRIPTION
**Pull request by man-shu**
For testing the new version of reporting on Travis CI, I have created a shorter version of the [Haxby example](https://github.com/neurospin/pypreprocess/blob/7ad460d4e0985e261536b4c95dfc805d03f3c70f/examples/easy_start/nipype_preproc_spm_haxby.py) that runs all the preprocessing steps on just one of the subjects and then creates a report for this. But the problem here is that a complete run of pypreprocess would require SPM and MATLAB installation which does not happen during Travis runs. 

Let me know if I have misunderstood what you were asking for, @bthirion.

Thanks!

### CODE REVIEW

File: pypreprocess/tests/test_reporting.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from nilearn.datasets import fetch_haxby
+from pypreprocess.nipype_preproc_spm_utils import (do_subjects_preproc,
+                                                    SubjectData)
+
+def test_reporting():
+    # fetch HAXBY dataset
+    haxby_data = fetch_haxby(subjects=1)
+
+    # set output dir
+    output_dir = os.path.join(os.path.dirname(haxby_data.mask),"haxby_runs")
```
Comment (bthirion) [inline pypreprocess/tests/test_reporting.py]:
```suggestion
    output_dir = os.path.join(os.path.dirname(haxby_data.mask), "haxby_runs")
```

**Review by bthirion — COMMENTED** [review]
Hm. Isn't there any way to set up a compiled spm in the travis environment ?

File: pypreprocess/tests/test_reporting.py (hunk 1 above)
Comment (swaythe) [reply pypreprocess/tests/test_reporting.py]: We could do the same thing done in circleci but SPM+Matlab installation takes forever. Maybe a pure python example instead of the SPM one?

Comment (bthirion) [reply pypreprocess/tests/test_reporting.py]: Agreed. Is it possible to call pure Python preprocessing instead ?

Comment (man-shu) [reply pypreprocess/tests/test_reporting.py]: It is just that the pure python preprocessing module does not use the new reporting yet. I only made changes in the spm module in the previous PR. But I could do that now. Should make those changes in this PR or would you rather prefer a separate PR for that?

**Comment by man-shu** [comment]
It is just that the pure python preprocessing module does not use the new reporting yet. I only made changes in the spm module in the previous PR. But I could do that now. Should make those changes in this PR or would you rather prefer a separate PR for that?

**Comment by man-shu** [comment]
I looked at pure python preprocessing implementation today, and there are a few issues with it:

- it does not implement segmentation and normalisation steps, both of which have specific reporting functions. So if we run a pure python example, it won't test the reporting at these steps.
- nevertheless, I tried using new reporting with pure python, and it looks like the log formats are different from those of spm, so logging would not be tested either.
-  format for coregistration output images also seems to be different here. So the plotting functions do not work with pure python coregistration. The previously written [pure python example](https://github.com/neurospin/pypreprocess/blob/7ad460d4e0985e261536b4c95dfc805d03f3c70f/examples/pure_python/pure_python_preproc_demo.py) also has `report` set to `False`, so it seems that the plotting was not written to work with pure python.

In conclusion, we are only left with the motion correction step, which is working right off the bat. Also, pure python preprocessing would require some more work so that it can use the new reporting. So should I start working towards that, @bthirion @swaythe?

**Comment by bthirion** [comment]
No, we should probably stop here.
Are the reports currently generated on circleci ?

**Comment by man-shu** [comment]
Yes, Circle CI runs all the easy start examples which generate reports and use SPM preprocessing.
